### ISSUE TITLE
Apple clang 15 compilation fails with: boost/container_hash/hash.hpp:130:33: error: no template named 'unary_function' in namespace 'std'

### ISSUE THREAD

**Issue by bhiogade** [vinecopulib/pyvinecopulib#116]
Trying to pip install pyvinecopulib on a vanilla macOS (Apple silicon), pip does not pick any binary wheel and starts compilation, which fails. The first error seems to be:
```
      lib/boost/boost/container_hash/hash.hpp:130:33: error: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
```
which might be related with the removal of `std::unary_function` from the C++ standard as of C++17. (see here: https://git.skewed.de/count0/graph-tool/-/issues/761).

System info:
```
% sw_vers 
ProductName:            macOS
ProductVersion:         14.2.1
BuildVersion:           23C71

% clang --version
Apple clang version 15.0.0 (clang-1500.1.0.2.5)
Target: arm64-apple-darwin23.2.0
```

Similar issue reported here: https://github.com/pingcap/tiflash/issues/5717

HTH

**Comment by tnagler** [comment]
Indeed, we have to replace our [prepackage boost headers](https://github.com/vinecopulib/pyvinecopulib/blob/main/lib/boost_1_71_0.tar.gz) with a newer version.

**Comment by tnagler** [comment]
fixed in #117

**Comment by tvatter** [comment]
Just to be clear, the issue is fixed in the dev branch, so @bhiogade will have to install from this one for now.
